File: companies/Hackerrank/decimal.java
import java.util.HashMap;
import java.util.Map;

class Main {

    public static String solve(String s) {
        s += " ";
        String delimiter = " ";

        String[] tokens = s.split(delimiter);
        Map<String, Integer> mp = new HashMap<>();
        mp.put("zero", 0);
        mp.put("one", 1);
        mp.put("two", 2);
        mp.put("three", 3);
        mp.put("four", 4);
        mp.put("five", 5);
        mp.put("six", 6);
        mp.put("seven", 7);
        mp.put("eight", 8);
        mp.put("nine", 9);

        Map<String, Integer> kmp = new HashMap<>();
        kmp.put("double", 2);
        kmp.put("triple", 3);

        StringBuilder res = new StringBuilder();
        int cur = 1;

        for (String token : tokens) {
            if (kmp.containsKey(token)) {
                cur = kmp.get(token);
            } else {
                while (cur-- > 0) {
                    res.append(mp.get(token));
                }
                cur = 1;
            }
        }

        return res.toString();
    }

    public static void main(String[] args) {
        System.out.println(solve("one two three double two one triple three one"));
    }
}

File: companies/Hackerrank/Decimal.cpp
#include <bits/stdc++.h>
#include "Decimal.h"

using namespace std;

string solve(string s){
    s +=" ";
    string delimiter = " ";

    size_t pos = 0;
    std::string token;
    map<string,int> mp;
    mp["zero"] = 0;
    mp["one"] = 1;
    mp["two"] = 2;
    mp["three"] = 3;
    mp["four"] = 4;
    mp["five"] = 5;
    mp["six"] = 6;
    mp["seven"] = 7;
    mp["eight"] = 8;
    mp["nine"] = 9;
    map<string,int> kmp;

    kmp["double"] = 2;
    kmp["triple"] = 3;
    string res;
    int cur = 1;
    while ((pos = s.find(delimiter)) != std::string::npos) {
        token = s.substr(0, pos);
        if(kmp.count(token)) {
            cur = kmp[token];
        }
        else {
            while(cur-->0) {
                res += to_string(mp[token]);
            }            
            cur= 1;
        }

        s.erase(0, pos + delimiter.length());
        
    }
    return res;
}
int main(int argc, char const *argv[])
{
    cout << solve("one two three double two one triple three one");
    return 0;
}
